operator/cluster/cluster.go

```diff
@@ -202,6 +190,56 @@ func AddClusterBase(clientset *kubernetes.Clientset, client *rest.RESTClient, cl
 
 }
 
+//getOrCreatePVC will first verify whether the pgcluster name matches the "current primary" CRD value of the pgcluster CRD.
+//Based on this information, the underlying PVC name for the primary pod can be validated. If the expected PVC exists,
+//the correct PVC name is returned.
+//If the "current primary" label and the pgcluster name do match, this is either the recreation of a cluster that has not
+//previously failed over, or it is a new pgcluster. If a recreation, the PVC is reused, otherwise a new PVC is allocated.
+func getOrCreatePVC(clientset *kubernetes.Clientset, cl *crv1.Pgcluster, namespace string) string {
```

Comment:
I think this should return `(string, error)` so the error gets bubbled up to the higher function.